plugins/plugins/consumer_restriction/config.go

```diff
@@ -50,9 +55,16 @@ func (conf *config) Init(cb api.ConfigCallbackHandler) error {
 		conf.allow = true
 	}
 
-	conf.consumers = make(map[string]struct{}, len(rules.Rules))
+	conf.consumers = make(map[string]*Rule, len(rules.Rules))
 	for _, r := range rules.Rules {
-		conf.consumers[r.Name] = struct{}{}
+		methods := make(map[string]bool)
+		for _, method := range r.GetMethods() {
+			methods[method] = true
```

Comment:
Maybe we can add a validation to make sure the method is uppercase?
https://github.com/bufbuild/protoc-gen-validate?tab=readme-ov-file#repeated

Reply:
> Maybe we can add a validation to make sure the method is uppercase? https://github.com/bufbuild/protoc-gen-validate?tab=readme-ov-file#repeated

add a validation in types/plugins/consumer_restriction/config.proto